src/Loader/FileGetContentsWebLoader.php

```diff
@@ -23,28 +23,32 @@
      * @param string               $prefix
      * @param JsonDecoderInterface $jsonDecoder
      */
-    public function __construct($prefix, JsonDecoderInterface $jsonDecoder = null)
+    public function __construct($prefix, DecoderManager $decoders = null)
     {
         $this->prefix      = $prefix;
-        $this->jsonDecoder = $jsonDecoder ?: new JsonDecoder();
+        $this->decoders = $decoders ?: new DecoderManager();
     }
 
     /**
      * {@inheritdoc}
      */
-    public function load($path)
+    public function load($path, $defaultExtension = 'json')
     {
-        $uri = $this->prefix . $path;
-        set_error_handler(function () use ($uri) {
-            throw SchemaLoadingException::create($uri);
-        });
+        $uri       = $this->prefix . $path;
+        $extension = isset(pathinfo($path)['extension']) ? pathinfo($path)['extension'] : $defaultExtension;
+
+        set_error_handler(
+            function () use ($uri) {
+                throw SchemaLoadingException::create($uri);
+            }
+        );
         $response = file_get_contents($uri);
         restore_error_handler();
 
         if (!$response) {
             throw SchemaLoadingException::create($uri);
         }
 
-        return $this->jsonDecoder->decode($response);
+        return $this->decoders->getDecoder($extension)->decode($response);
```

Comment:
It doesn't seem like the web loader should be using the file extension to determine what decoder to use.

Schemas can be hosted at URIs that do not have a file extension but do return a `Content-Type` header.  In that case it would make more sense to use the content-type to determine which decoder to use.